extensions/ql-vscode/src/query-history.ts

```diff
@@ -933,6 +935,8 @@ export class QueryHistoryManager extends DisposableObject {
       }
     } else if (finalSingleItem.t === 'remote') {
       externalFilePath = path.join(this.queryStorageDir, finalSingleItem.queryId, 'timestamp');
+    } else if (finalSingleItem.t === 'variant-analysis') {
+      externalFilePath = path.join(this.variantAnalysisManager.storagePath, finalSingleItem.variantAnalysis.id.toString(), 'timestamp');
```

Comment:
We don't yet create a `timestamp` file for variant analysis items, so we always fall back to the block [below](https://github.com/github/vscode-codeql/pull/1621/files#diff-9a58bafccb68e723c78ef4526185ed737f27b29330407b288eb2dbf682256332R48):
```ts
if (!(await fs.pathExists(externalFilePath))) {
  // timestamp file is missing (manually deleted?) try selecting the parent folder.
  // It's less nice, but at least it will work.
  externalFilePath = path.dirname(externalFilePath);
```

This works fine, but should we be creating a `timestamp` file? (see https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/helpers.ts#L560-L564)

Reply:
I don't have full context but seems like we probably should create a timestamp file for new variant analyses. Seems we do this in the remote queries case by calling `await this.prepareStorageDirectory(queryId);` just before we trigger the monitoring command.

But as you explain, this is semi-independent of this PR because there's a fallback that covers us. So I'd suggest we add thetimestamp file but it doesn't necessarily have to be in this PR or block this PR.